lib/src/switch/Switch.tsx

```diff
@@ -51,73 +48,59 @@ const DxcSwitch = ({
     if (checked === undefined) {
       const isChecked = event.target.checked ?? !innerChecked;
       setInnerChecked(isChecked);
-      if (typeof onChange === "function") {
-        onChange(isChecked);
-      }
+      onChange?.(isChecked);
     } else {
-      if (typeof onChange === "function") {
-        onChange(!checked);
-      }
+      onChange?.(!checked);
     }
   };
 
   return (
     <ThemeProvider theme={colorsTheme.switch}>
-      <SwitchContainer margin={margin} size={size} onKeyDown={handleOnKeyDown}>
-        {labelPosition === "before" && hasLabel && (
+      <SwitchContainer
+        margin={margin}
+        size={size}
+        onKeyDown={handleOnKeyDown}
+        disabled={disabled}
+        onClick={!disabled ? handlerSwitchChange : undefined}
+      >
+        {labelPosition === "before" && label && (
           <LabelContainer
             id={labelId}
             labelPosition={labelPosition}
-            onClick={!disabled ? handlerSwitchChange : undefined}
             disabled={disabled}
             backgroundType={backgroundType}
-            hasLabel={hasLabel}
+            label={label}
           >
             {label} {optional && <>{translatedLabels.formFields.optionalLabel}</>}
           </LabelContainer>
         )}
-        <SwitchBase
-          labelPosition={labelPosition}
-          hasLabel={hasLabel}
-          htmlFor={labelId}
-          onClick={disabled === true ? () => {} : handlerSwitchChange}
-        >
-          <SwitchInput
+        <SwitchBase>
+          <ValueInput
             type="checkbox"
-            role="switch"
             name={name}
-            id={labelId}
-            disabled={disabled}
+            aria-hidden={true}
             value={value}
-            aria-labelledby={labelId}
-            aria-label={hasLabel ? label : undefined}
+            disabled={disabled}
+            readOnly
+          ></ValueInput>
```

Comment:
Remove the closing tag `</ValueInput>.` When a component doesn't have children the common code recommendation is to close it within the opening tag (`<Component />`)

Reply:
Done